src/Unleash/DefaultUnleash.cs

```diff
@@ -124,14 +126,31 @@ private bool CheckIsEnabled(string toggleName, UnleashContext context, bool defa
             }
             else
             {
-                enabled = featureToggle.Strategies.Any(s => GetStrategyOrUnknown(s.Name).IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints)));
+                strategy = featureToggle.Strategies
+                    .FirstOrDefault(s =>
+                        GetStrategyOrUnknown(s.Name)
+                        .IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints))
+                    );
+                enabled = strategy != null;
+            }
+
+            if (strategy != null)
+            {
+                string groupId = null;
+                strategy.Parameters.TryGetValue("groupId", out groupId);
+                variant = enabled ? VariantUtils.SelectVariant(groupId, context, strategy.Variants) : null;
```

Comment:
If strategy does not have group id , what happens? It should default to using feature name as groupId